compiler/haskell-ide-core/src/Development/IDE/Functions/Compile.hs

```diff
@@ -222,25 +222,42 @@ compileModule opt mod packageState uniqSupply deps pkgs tmr =
 runGhcSessionExcept
     :: IdeOptions
     -> Maybe ParsedModule
-    -> PackageDynFlags
+    -> HscEnv
     -> Ex.ExceptT e Ghc a
     -> Ex.ExceptT e IO a
 runGhcSessionExcept opts mbMod pkg m =
     Ex.ExceptT $ runGhcSession opts mbMod pkg $ Ex.runExceptT m
 
 
-getGhcDynFlags :: IdeOptions -> ParsedModule -> PackageDynFlags -> IO DynFlags
+getGhcDynFlags :: IdeOptions -> ParsedModule -> HscEnv -> IO DynFlags
 getGhcDynFlags opts mod pkg = runGhcSession opts (Just mod) pkg getSessionDynFlags
 
 -- | Evaluate a GHC session using a new environment constructed with
 -- the supplied options.
 runGhcSession
     :: IdeOptions
     -> Maybe ParsedModule
-    -> PackageDynFlags
+    -> HscEnv
     -> Ghc a
     -> IO a
-runGhcSession IdeOptions{..} = optRunGhcSession
+runGhcSession IdeOptions{..} modu env act = runGhcEnv env $ do
+    modifyDynFlags $ \x -> x{importPaths = maybe [] moduleImportPaths modu ++ importPaths x}
```

Comment:
Should the paths be de-duplicated? 
(Previously the paths were added in the same way but probably forgotten afterwards)

Reply:
I don't think repeating them is harmful, so inclined to not bother. Can you think of a potential problem if they aren't dedupe'd?

Reply:
Note that these don't accumulate in the same HscEnv, since we create a newIORef for each session, so they won't get worse over time.

Reply:
I'll put it in a future commit.